Code/autopkglib/SparkleUpdateInfoProvider.py

```diff
@@ -118,27 +125,88 @@ class SparkleUpdateInfoProvider(Processor):
-                for header, value in headers.items():
-                    cmd.extend(["--header", "%s: %s" % (header, value)])
-            cmd.append(url)
-            proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-            (data, stderr) = proc.communicate()
-            if proc.returncode:
-                raise ProcessorError("Could not retrieve URL %s: %s" % (url, stderr))
-        except OSError:
+            content = super(SparkleUpdateInfoProvider, self).download(curl_cmd)
+        except ProcessorError:
             raise ProcessorError("Could not retrieve URL: %s" % url)
 
-        return data
+        return content
 
     def get_feed_data(self, url):
+        """Downloads raw feed XML"""
+
+        # query string
+        if "appcast_query_pairs" in self.env:
+            queries = self.env["appcast_query_pairs"]
+            new_query = urllib.urlencode([(k, v) for (k, v) in queries.items()])
+            scheme, netloc, path, _, frag = urlparse.urlsplit(url)
+            url = urlparse.urlunsplit((scheme, netloc, path, new_query, frag))
+
+        data = self.fetch_content(url, headers=self.env.get("appcast_request_headers"))
+        return data
+
+    def build_url(self, enclosure):
+        """URL-quote the path component to handle spaces, etc.
+        (Panic apps do this)"""
+
+        url_bits = urlparse.urlsplit(enclosure.get("url"))
+        if self.env.get("urlencode_path_component", True):
+            encoded_path = urllib.quote(url_bits.path)
+        else:
+            encoded_path = url_bits.path
+        built_url = url_bits.scheme + "://" + url_bits.netloc + encoded_path
+        if url_bits.query:
+            built_url += "?" + url_bits.query
+        return built_url
+
+    def determine_version(self, enclosure, url):
+        """Gets version from enclosure"""
+
+        version = enclosure.get("{%s}version" % self.xmlns)
+
+        if version is None:
+            # Sparkle tries to guess a version from the download URL for
+            # rare cases where there is no sparkle:version enclosure
+            # attribute, for the format: AppnameOrAnything_1.2.3.zip
+            # https://github.com/sparkle-project/Sparkle/blob/
+            # 89081ca030c0de218400f7c0f97530df524d687d/Sparkle/
+            # SUAppcastItem.m#L69-L76
+            #
+            # We can even support OmniGroup's alternate appcast format
+            # by cheating and using the '-' as a delimiter to derive
+            # version info
+            filename = os.path.basename(os.path.splitext(url)[0])
+            for delimiter in ["_", "-"]:
+                if delimiter in filename:
+                    version = filename.split(delimiter)[-1]
+                    break
+        # if we still found nothing, fail
+        if version is None:
+            raise ProcessorError("Can't extract version info from item in feed!")
+        return version
+
+    def parse_feed_data(self, data):
```

Comment:
Split from `get_feed_data()` some of the [lines](https://github.com/autopkg/autopkg/blob/15c6307254d4ef1d1dbb8a2648a204a340f5b967/Code/autopkglib/SparkleUpdateInfoProvider.py#L170) in range 170..233